neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/dialect/impl/RemoteNeo4jSequenceGenerator.java

```diff
@@ -0,0 +1,253 @@
+		queryBuilder.append( " { " );
+		queryBuilder.append( idSourceKeyMetadata.getKeyColumnName() );
+		queryBuilder.append( ": {" );
+		queryBuilder.append( SEQUENCE_NAME_QUERY_PARAM );
+		queryBuilder.append( "}} ) " );
+		queryBuilder.append( " ON MATCH SET n.__locked=true " );
+		queryBuilder.append( " ON CREATE SET n.__locked=true, n." );
+		queryBuilder.append( idSourceKeyMetadata.getValueColumnName() );
+		queryBuilder.append( " = " );
+		queryBuilder.append( initialValue );
+		queryBuilder.append( " RETURN n." );
+		queryBuilder.append( idSourceKeyMetadata.getValueColumnName() );
+		String query = queryBuilder.toString();
+		return query;
+	}
+
+	private void getUpdateTableSequenceQuery(Statements statements, IdSourceKeyMetadata idSourceKeyMetadata, String sequenceName, int initialValue, int increment) {
+		String acquireLockQuery = acquireLockQuery( idSourceKeyMetadata, initialValue, increment );
+		Statement acquireLockStatement = new Statement( acquireLockQuery, params( sequenceName, initialValue, increment ) );
+		acquireLockStatement.setResultDataContents( Arrays.asList( Statement.AS_ROW ) );
+		statements.addStatement( acquireLockStatement );
+
+		String updateQuery = increaseQuery( idSourceKeyMetadata, increment );
+		Statement updateStatement = new Statement( updateQuery, params( sequenceName, initialValue, increment ) );
+		updateStatement.setResultDataContents( Arrays.asList( Statement.AS_ROW ) );
+		statements.addStatement( updateStatement );
+	}
+
+	protected String increaseQuery(IdSourceKeyMetadata idSourceKeyMetadata, int increment) {
+		StringBuilder queryBuilder = new StringBuilder();
+		Label generatorKeyLabel = DynamicLabel.label( idSourceKeyMetadata.getName() );
+		queryBuilder.append( " MATCH (n " );
+		queryBuilder.append( labels( generatorKeyLabel.name(), NodeLabel.TABLE_BASED_SEQUENCE.name() ) );
+		queryBuilder.append( " { " );
+		queryBuilder.append( idSourceKeyMetadata.getKeyColumnName() );
+		queryBuilder.append( ": {" );
+		queryBuilder.append( SEQUENCE_NAME_QUERY_PARAM );
+		queryBuilder.append( "}} )" );
+		queryBuilder.append( " SET n." );
+		queryBuilder.append( idSourceKeyMetadata.getValueColumnName() );
+		queryBuilder.append( " = n." );
+		queryBuilder.append( idSourceKeyMetadata.getValueColumnName() );
+		queryBuilder.append( " + " );
+		queryBuilder.append( increment );
+		queryBuilder.append( " REMOVE n.__locked RETURN n." );
+		queryBuilder.append( idSourceKeyMetadata.getValueColumnName() );
+		String query = queryBuilder.toString();
+		return query;
+	}
+
+	/**
+	 * Generate the next value in a sequence for a given {@link IdSourceKey}.
+	 *
+	 * @param idSourceKey identifies the generator
+	 * @param increment the difference between to consecutive values in the sequence
+	 * @param initialValue the initial value of the given generator
+	 * @return the next value in a sequence
+	 */
+	public Number nextValue(IdSourceKey idSourceKey, int increment, int initialValue) {
```

Comment:
The parameters increment and initialValue are interverted between the embedded and the remote versions. Doesn't look like a good idea. Maybe having an abstract method in the parent class would be handy to avoid this?